### DISCUSSION TITLE
ExtractSubset() from Paraview Save State not defined in Trame

### DISCUSSION THREAD

**Discussion by staceywhitmore-inl** [Kitware/trame#537] [Q&A] [answered]
I have a series of Tiffs I am extracting a subset from in Paraview Desktop using the Filters > Common > ExtractSubset filter. After entering the range on K(i.e., Z) axis and click apply I set the representation back to Volume and click close on the eye in the originally imported dataset to just show what I have extracted. 

Then I save this state as a .py file to see the methods I will need to call in my Trame app to achieve this same result. From that I can see that I need to call the following in my Trame app

```
  extractSubset1 = simple.ExtractSubset(registrationName='ExtractSubset1', Input=tiffObj) 
  extractSubset1.VOI = [0, 1535, 0, 1023, 30, 40] # extract indices 30 — 40
```
  
  *Where tiffObj is what I have successfully been able to render by previously calling
`  tiffObj = simple.TIFFSeriesReader()`
  
  When I try to serve my app I get the following error in my console
`  NameError: name 'ExtractSubset' is not defined
`  
Is there an import I am missing in pvpython to use the `ExtractSubset()` method? I've imported the same imports saved to the state.py exported from Paraview and can't find any info on this method here. (Perhaps, somebody could show me where I would look for documentation on this method.)

I am using the following imports
```
# from tkinter import filedialog, Tk #Does not work
import os
from trame_server.utils.hot_reload import hot_reload
import paraview.web.venv  
from trame.app import get_server
from trame.decorators import TrameApp #, change
from trame.widgets import vuetify, paraview, trame, html
from trame.ui.vuetify import SinglePageWithDrawerLayout
from paraview import simple
```

**Comment by jourdain** [comment, marked as answer]
It should work unless you don't have the same ParaView version between the trace and the trame one.

**Reply by staceywhitmore-inl** [reply]
YES. Not sure what I was doing wrong before but **_IT'S WORKING NOW_**!
My bad.
(I'm using Paraview 5.11.12 in both. Perhaps I would be better to upgrade them both to 5.12 just to be up-to-date, no?)

Now I just need to figure out how to change the extracted subset range using a VSlider in get the view to update in real time...
